analysis/src/Xform.ml

```diff
@@ -249,14 +271,133 @@ module AddTypeAnnotation = struct
               ( rangeOfLoc locItem.loc,
                 "(" ^ name ^ ": " ^ (typ |> Shared.typeToString) ^ ")" )
           in
+          let uri = Uri.fromPath path |> Uri.toString in
           let codeAction =
             CodeActions.make ~title:"Add type annotation" ~kind:RefactorRewrite
-              ~uri:path ~newText ~range
+              ~edit:
+                {
+                  documentChanges =
+                    [
+                      TextDocumentEdit
+                        {
+                          textDocument = {version = None; uri};
+                          edits = [{newText; range}];
+                        };
+                    ];
+                }
           in
           codeActions := codeAction :: !codeActions
         | _ -> ()))
 end
 
+module TypeToModule = struct
+  (* Convert a type it into its own submodule *)
+  let mkIterator ~pos ~result ~newTypeName =
+    let changeTypeDecl (typ : Parsetree.type_declaration) ~txt =
+      match typ.ptype_manifest with
+      | None ->
+        Ast_helper.Type.mk
+          {txt; loc = typ.ptype_name.loc}
+          ~loc:typ.ptype_loc ~attrs:typ.ptype_attributes
+          ~params:typ.ptype_params ~cstrs:typ.ptype_cstrs ~kind:typ.ptype_kind
+          ~priv:typ.ptype_private
+      | Some manifest ->
+        Ast_helper.Type.mk
+          {txt; loc = typ.ptype_name.loc}
+          ~loc:typ.ptype_loc ~attrs:typ.ptype_attributes
+          ~params:typ.ptype_params ~cstrs:typ.ptype_cstrs ~kind:typ.ptype_kind
+          ~priv:typ.ptype_private ~manifest
+    in
```

Comment:
This function is to change the name of type declaration. Is there another way to do this?

`typ.ptype_manifest` is a option, but `Ast_helper.Type.mk` `~manifest` not expected an option